src/pages/tasks/TaskShareDestinationSelectorModal.js

```diff
@@ -1,154 +1,103 @@
-    personalDetails: PropTypes.objectOf(personalDetailsPropType),
-
     /** All reports shared with the user */
     reports: PropTypes.objectOf(reportPropTypes),
-
-    /** Whether we are searching for reports in the server */
+    /** Whether or not we are searching for reports on the server */
     isSearchingForReports: PropTypes.bool,
-
-    ...withLocalizePropTypes,
 };
 
 const defaultProps = {
-    betas: [],
-    personalDetails: {},
     reports: {},
     isSearchingForReports: false,
 };
 
-function TaskShareDestinationSelectorModal(props) {
-    const styles = useThemeStyles();
-    const [searchValue, setSearchValue] = useState('');
-    const [headerMessage, setHeaderMessage] = useState('');
-    const [filteredRecentReports, setFilteredRecentReports] = useState([]);
+const selectReportHandler = (option) => {
+    if (!option || !option.reportID) {
+        return;
+    }
 
-    const {inputCallbackRef} = useAutoFocusInput();
-    const {isSearchingForReports} = props;
-    const {isOffline} = useNetwork();
+    Task.setShareDestinationValue(option.reportID);
+    Navigation.goBack(ROUTES.NEW_TASK);
+};
 
-    const filteredReports = useMemo(() => {
-        const reports = {};
-        _.keys(props.reports).forEach((reportKey) => {
-            if (
-                !ReportUtils.canUserPerformWriteAction(props.reports[reportKey]) ||
-                !ReportUtils.canCreateTaskInReport(props.reports[reportKey]) ||
-                ReportUtils.isCanceledTaskReport(props.reports[reportKey])
-            ) {
-                return;
+const reportFilter = (reports) =>
+    reduce(
+        keys(reports),
+        (filtered, reportKey) => {
+            const report = reports[reportKey];
+            if (ReportUtils.canUserPerformWriteAction(report) && ReportUtils.canCreateTaskInReport(report) && !ReportUtils.isCanceledTaskReport(report)) {
+                return {...filtered, [reportKey]: report};
             }
-            reports[reportKey] = props.reports[reportKey];
-        });
-        return reports;
-    }, [props.reports]);
-    const updateOptions = useCallback(() => {
-        const {recentReports} = OptionsListUtils.getShareDestinationOptions(filteredReports, props.personalDetails, props.betas, searchValue.trim(), [], CONST.EXPENSIFY_EMAILS, true);
-
-        setHeaderMessage(OptionsListUtils.getHeaderMessage(recentReports?.length !== 0, false, searchValue));
-
-        setFilteredRecentReports(recentReports);
-    }, [props, searchValue, filteredReports]);
-
-    useEffect(() => {
-        const debouncedSearch = _.debounce(updateOptions, 150);
-        debouncedSearch();
-        return () => {
-            debouncedSearch.cancel();
-        };
-    }, [updateOptions]);
+            return filtered;
+        },
+        {},
+    );
 
-    const getSections = () => {
-        const sections = [];
-        let indexOffset = 0;
+function TaskShareDestinationSelectorModal({reports, isSearchingForReports}) {
```

Comment:
The tash share page has a stuttering animation.

https://github.com/Expensify/App/assets/29673073/fe79ca98-d4b6-4f22-bf87-ab6674dba9d4